packages/flutter/lib/src/widgets/_web_browser_detection.dart

```diff
@@ -0,0 +1,11 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui_web' as ui_web;
+
+/// A class that provides information about the current browser when on the web.
+class WebBrowserDetection {
+  /// Whether the current browser is webkit (Safari).
+  static bool get browserIsSafari => ui_web.BrowserDetection.instance.isSafari;
```

Comment:
I think this would be the first time we are using BrowserDetection in the framework? Maybe we should check with someone from the web team before we do this.

Makes sense that we need to do this in order to get the styling right, though.

Reply:
CC @mdebbar from triage

Can we do feature detection?

Reply:
I misunderstood the PR during triage. It looks like the purpose of the PR is to choose a different for specific platforms, so checking for `iOS` and/or `isSafari` is the way to go here.

Feature detection is for when you need to use a certain feature that you know only exists in a specific browser. It's better to check the existence of the feature at runtime, rather than just check for what browser you're running on.